src/pip/_internal/commands/__init__.py

```diff
@@ -3,57 +3,117 @@
 """
 from __future__ import absolute_import
 
-from pip._internal.commands.completion import CompletionCommand
-from pip._internal.commands.configuration import ConfigurationCommand
-from pip._internal.commands.debug import DebugCommand
-from pip._internal.commands.download import DownloadCommand
-from pip._internal.commands.freeze import FreezeCommand
-from pip._internal.commands.hash import HashCommand
-from pip._internal.commands.help import HelpCommand
-from pip._internal.commands.list import ListCommand
-from pip._internal.commands.check import CheckCommand
-from pip._internal.commands.search import SearchCommand
-from pip._internal.commands.show import ShowCommand
-from pip._internal.commands.install import InstallCommand
-from pip._internal.commands.uninstall import UninstallCommand
-from pip._internal.commands.wheel import WheelCommand
+import importlib
 
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
 if MYPY_CHECK_RUNNING:
-    from typing import List, Type
+    from typing import Any, Dict, Iterable, List, Tuple
     from pip._internal.cli.base_command import Command
 
+
+commands_dict = {
+    'check': (
+        'pip._internal.commands.check', 'CheckCommand',
+        'Verify installed packages have compatible dependencies.',
+    ),
+    'completion': (
+        'pip._internal.commands.completion', 'CompletionCommand',
+        'A helper command used for command completion.',
+    ),
+    'config': (
+        'pip._internal.commands.configuration', 'ConfigurationCommand',
```

Comment:
It might even make sense to, in a follow up, rename the this module to "config" and then simply reuse the key as the module name, since that's the case for everything else.

Reply:
Good thought!